jdaviz/core/tools.py

```diff
@@ -93,12 +88,32 @@ def deactivate(self):
     def on_limits_change(self, *args):
         # from_lims: limits in the viewer belonging to the tool
         from_lims = {k: getattr(self.viewer.state, k) for k in self.match_keys}
+        orig_refdata = self.viewer.state.reference_data
 
         for viewer in self._iter_matched_viewers(include_self=False):
             # orig_lims: limits in this "matched" viewer
             # to_lims: proposed new limits for this "matched" viewer
             orig_lims = {k: getattr(viewer.state, k) for k in self.match_keys}
             to_lims = self._map_limits(self.viewer, viewer, from_lims)
+
+            to_refdata = viewer.state.reference_data
+
+            if orig_refdata != to_refdata and hasattr(self.viewer, '_get_fov'):
+                # if the viewers have different reference data,
+                # rescale the zoom and center allowing for different
+                # viewer rotations:
+                orig_fov_sky = self.viewer._get_fov(orig_refdata)
+                to_fov_sky = viewer._get_fov(orig_refdata)
+                sky_cen = self.viewer._get_center_skycoord(orig_refdata)
+
+                viewer.zoom(
+                    float(to_fov_sky / orig_fov_sky)
+                )
+                viewer.center_on(sky_cen)
```

Comment:
are these processed sequentially?  If they're just modifying the limits under the hood, does wrapping in `delay_callback` work/help?

Reply:
I didn't expect it to matter or make a difference.